File: C/263A-Beautiful Matrix.c
/*
    Solution to Codeforces problem 263A
    Copyright (c) GeneralMing. All rights reserved.

    https://github.com/GeneralMing/codeforces
*/

#include <stdio.h>
 
int main()
{
    int a[5][5], x, y, lock = 1, count = 0;
    for(int i = 0; i < 5; i++)
        for(int j = 0; j < 5; j++)
        {
            scanf("%d", &a[i][j]);
            if(a[i][j] == 1)
            {
                x = i;
                y = j;
            }
        }
    while(lock)
    {
        lock = 0;
        if(x > 2)
        {
            count++;
            x--;
            lock = 1;
        }
        if(x < 2)
        {
            count++;
            x++;
            lock = 1;
        }
        if(y > 2)
        {
            count++;
            y--;
            lock = 1;
        }
        if(y < 2)
        {
            count++;
            y++;
            lock = 1;
        }
    }
    printf("%d\n", count);
    return 0;
}

File: Python/263A-Beautiful Matrix.py
"""
    Solution to Codeforces problem 263A
    Copyright (c) GeneralMing. All rights reserved.

    https://github.com/GeneralMing/codeforces
"""

a = []
count = 0
lock = True
for i in range(5):
    a.append(input().split(" "))

for i in range(5):
    for j in range(5):
        a[i][j] = int(a[i][j])
        if(a[i][j] == 1):
            x = i
            y = j

while(lock):
    lock = False;
    if(x > 2):
        count += 1;
        x -= 1;
        lock = True;
    if(x < 2):
        count += 1;
        x += 1;
        lock = True;
    if(y > 2):
        count += 1;
        y -= 1;
        lock = True;
    if(y < 2):
        count += 1;
        y += 1;
        lock = True;
print(count)
    